### ISSUE TITLE
Replace extract-xiso with a different tool

### ISSUE THREAD

**Issue by JayFoxRox** [XboxDev/extract-xiso#12]
**This is discussion. Not a plan.**

With all those PRs pouring in (which remove legacy features or bad code), I'm wondering if it's such a good idea to even maintain extract-xiso.
Aside for creating functional ISOs for XQEMU, nothing really seems to work.

It can obviously not create redump XISO (no tool can) and it can't even extract them.

mborgerson has mentioned that extract-xiso could be replaced with a short python script - and I agree.
There's also other projects which seem to do a similiar, or even better job.

Yesterday I found myself with my redump XISO dumps and wanted to run a game on real Xbox. As I'm on Linux I had a hard time finding any software to deal with that.
I'd expect extract-xiso to support this, but it doesn't. While looking around, someone on redump IRC recommended xbiso. I've used the [xbiso C++ rewrite](https://github.com/OpenXDK/xbiso) by @thrimbor ~~which worked just fine for extracting redump XISOs~~. It might work fine, but [I used a custom branch which has a hack](https://github.com/JayFoxRox/xbiso/pull/1/files#r209454444) (not sure if relevant).

I'm not sure how good the code quality of xbiso is (despite having hacked together a version to export metadata about XISO in late 2017).
In my opinion we should review tools like it (what is supported? is there feature creep?) and find one which could possibly replace extract-xiso if necessary.

**Thoughts?**

- Should we keep extract-xiso?
- Move xbiso to XboxDev?
- New tool from scratch?
...

**Comment by mborgerson** [comment]
The code for extract-xiso is not very easy to follow which of course stunts maintainability, but the tool itself does work well (at least in my experience). I don't foresee a lot of changes needed going forward once all build issues are resolved, but that's just speculation. If someone wants to clean the code up a bit (refactoring, cleaning up some of the error handling, adding comments), that would be nice and I'm open to reviewing PRs.

As xbiso does not actually support creating ISOs yet, I don't think it's a suitable candidate to outright supplant extract-xiso at this time...but if someone else wants to develop and maintain xbiso, or a new tool (preferably in Python), which can create/extract, I would support that.

**Comment by JayFoxRox** [comment]
I had actually prototyped a tool in Python after the debacle in #25.

That code can be found here: https://github.com/JayFoxRox/extract-xiso/pull/2.

**Comment by Natetronn** [comment]
I realize this is an old post and @JayFoxRox, maybe you're not longer on Linux anymore or have found something better but, I just wanted to point out that [xbfuse](https://github.com/multimediamike/xbfuse) can be used on Linux along with extraxt-xiso for creating functional ISOs for XQEMU:

**Comment by Leadhood** [comment]
@JayFoxRox "Yesterday I found myself with my redump XISO dumps and wanted to run a game on real Xbox. As I'm on Linux I had a hard time finding any software to deal with that."

If you are on linux you can just use DD in your terminal:

dd if=game.iso of=xiso.iso skip=387k bs=1k

**Comment by dracc** [comment]
@Leadhood: Yes, I'm sure JFR is [aware of that workaround](https://github.com/XboxDev/extract-xiso/issues/32#issuecomment-545213567). But it is just that, a workaround - not a fix.

**Comment by mborgerson** [comment]
@antangelo has recently developed [xdvdfs](https://github.com/antangelo/xdvdfs) command line tool and library for xiso management. There's also [a nice web interface](https://xiso.antangelo.com/) for it.

Now that we have at least one viable alternative with read/write functionality, I'm in favor of dropping support for extract-xiso unless another maintainer wants to continue development and review/merge patches.

**Comment by mborgerson** [comment]
I think this issue can be closed now (see last comment)

**Comment by mborgerson** [comment]
Actually, I'll leave it open until we update the README with a deprecation notice and a link to something else (like xdvdfs).
